config/jest/SetupSUC.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import initSonarUICommon from '../../helpers/init';
+
+initSonarUICommon({ l10nLocale: 'en', l10nMessages: {}, urlContext: '' });
```

Comment:
This is fine but I would like to know if you thought about an API like this:
```
initSonarUICommon({ urlContext: '' })
  .configureMessages({locale: 'en', messages: {}}); 
```
I don't think it would bring any value now, but depending on how we want to make sonar-ui-common configurable it would be nice to use something to kind of "namespace" the configuration object into the API. 
That would also help removing the `l10n` noise I mention in another comment.

Reply:
Thinking again about this, we could also inject the theme in some API.

Reply:
Yeah there is probably way to improve the API, we tried to keep it simple there. But maybe it's better to have separate initialization function. But I feel like the API you suggested might be overly complex for what we need, it would mean return a class instance that has init functions that would set a global variable ?

Regarding the theme:
> You can also see that the theme is not initialized by this function, we could have do some hacking like we do for the tests to initialize the theme through this function but that would break the React Context, because we would not use the context provider anymore and changes of the theme object would not be reflected anymore in the context consumer (we don't do that currently but we prefer avoid surprises for the future).

Reply:
I'd be happy to hear about the API from @jeremy-davis-sonarsource and @philippe-perrin-sonarsource too before we change something

Reply:
Simple is good ! :)
Yeah it is more complex, it would bring a class with methods that return the instance etc...
I'll just repeat one thing from my previous message: "I don't think it would bring any value now", there is no need to change anything but we can think about it during a Web coffee break !

Reply:
I like the dissociation of the urlContext from the locale & messages. Maybe we could expose a plain object with references to individual functions ?
The setters could return the object, for chaining:

```
import initializer from 's-u-c'

initializer.setUrlContext('/');
initializer
  .setMessages({...})
  .setLocale('en');

```

Reply:
I like this design too, I'll give a try!